pkgs/by-name/xf/xfox/package.nix

```
@@ -0,0 +1,32 @@
+{
+  python3Packages,
+  fetchFromGitHub,
+  lib,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "xfox";
+  version = "1.0.0";
+  pyproject = true;
+  src = fetchFromGitHub {
+    owner = "pxlman";
+    repo = "xfox";
+    tag = "v${version}";
+    sha256 = "sha256-gbcJ02ZaPqiCo0VHnVOYzrWR00dQXgKavFTgP+6rS34=";
+  };
+  propagatedBuildInputs = [
+    python3Packages.evdev
+    python3Packages.libevdev
+    python3Packages.setuptools
+  ];
```

1. `setuptools` (or any other Python builder) is supposed to be executed, thus it should go to `build-system` (same as `nativeBuildInputs`, but for Python packages)
2. and inputs in Pyproject recipes are called `build-system` and `dependencies` (instead of `nativeBuildInputs` and `propagatedBuildInputs`), just like in `pyproject.toml`. The original attrs can be used for non-Python dependencies tho, like Pandoc to generate manpages or some setup hook from Nixpkgs
```suggestion
  build-system = [ python3Packages.setuptools ];
  dependencies = [
    python3Packages.evdev
    python3Packages.libevdev
  ];
```
(if for some reason you `import setuptools` somewhere in your code, you can add it back to `dependencies`)